R/createmodelutils.R

```diff
@@ -5,20 +5,16 @@
 #' @return list of x, y, and weights with respective names
 #' @export
 getNamesFromOrdered <- function(use_weights, data_names) {
-  name_y_var <- data_names[1]
-  if(use_weights){
-    names_x_var <- data_names[2:length(data_names)-1]
-    name_weights_var <- data_names[length(data_names)]
-  } else {
-    names_x_var <- data_names[2:length(data_names)]
-    name_weights_var <- NULL
+  minimum_fields <- 2 + use_weights
+  assertthat::assert_that(length(data_names) >= minimum_fields)
```

Comment:
That's a fair point. I will instead assert classes on x, y, w which will stop the opportunity that they are null and be more robust in general I think.